src/app/rpmostree-compose-builtin-tree.c

```diff
@@ -849,107 +956,39 @@ impl_compose_tree (const char      *treefile_pathstr,
         }
     }
 
-  g_autoptr(GPtrArray) packages = g_ptr_array_new_with_free_func (g_free);
-
-  if (json_object_has_member (treefile, "bootstrap_packages"))
-    {
-      if (!_rpmostree_jsonutil_append_string_array_to (treefile, "bootstrap_packages", packages, error))
-        return FALSE;
-    }
-  if (!_rpmostree_jsonutil_append_string_array_to (treefile, "packages", packages, error))
-    return FALSE;
-
-  { g_autofree char *thisarch_packages = g_strconcat ("packages-", dnf_context_get_base_arch (rpmostree_context_get_hif (self->corectx)), NULL);
-
-    if (json_object_has_member (treefile, thisarch_packages))
-      {
-        if (!_rpmostree_jsonutil_append_string_array_to (treefile, thisarch_packages, packages, error))
-          return FALSE;
-      }
-  }
-  g_ptr_array_add (packages, NULL);
-
-  { glnx_unref_object JsonGenerator *generator = json_generator_new ();
-    char *treefile_buf = NULL;
-    gsize len;
-
-    json_generator_set_root (generator, treefile_rootval);
-    json_generator_set_pretty (generator, TRUE);
-    treefile_buf = json_generator_to_data (generator, &len);
-
-    self->serialized_treefile = g_bytes_new_take (treefile_buf, len);
-  }
-
-  /* Download rpm-md repos, packages, do install */
-  g_autofree char *new_inputhash = NULL;
-  { gboolean unmodified = FALSE;
-
-    if (!install_packages_in_root (self, treefile, rootfs_fd,
-                                   (char**)packages->pdata,
-                                   opt_force_nocache ? NULL : &unmodified,
-                                   &new_inputhash,
-                                   cancellable, error))
-      return FALSE;
-
-    if (unmodified)
-      {
-        g_print ("No apparent changes since previous commit; use --force-nocache to override\n");
-        /* Note early return */
-        return TRUE;
-      }
-    else if (opt_dry_run)
-      {
-        g_print ("--dry-run complete");
-        if (opt_touch_if_changed)
-          g_print (", updating --touch-if-changed=%s", opt_touch_if_changed);
-        g_print ("; exiting\n");
-        if (!process_touch_if_changed (error))
-          return FALSE;
-        /* Note early return */
-        return TRUE;
-      }
-  }
-
-  /* Destroy this now so the libdnf stack won't have any references
-   * into the filesystem before we manipulate it.
-   */
-  g_clear_object (&self->corectx);
-
-  if (g_strcmp0 (g_getenv ("RPM_OSTREE_BREAK"), "post-yum") == 0)
```

Comment:
Yes, good catch!  I always felt it was a hack...and one no longer necessary since basically one can just run `rpm-ostree compose install-root` and get that result.